odoo/addons/base/tests/test_module.py

```diff
@@ -0,0 +1,88 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import os.path
+import tempfile
+from os.path import join as opj
+from unittest.mock import patch
+
+import odoo.addons
+from odoo.modules.module import load_manifest
+from odoo.release import major_version
+from odoo.tests.common import BaseCase
+
+
+class TestModuleManifest(BaseCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_dir = tempfile.TemporaryDirectory(prefix='odoo-test-addons-')
+        cls.addClassCleanup(cls._tmp_dir.cleanup)
+        cls.addons_path = cls._tmp_dir.name
+
+        patcher = patch.object(odoo.addons, '__path__', [cls.addons_path])
+        patcher.start()
+        cls.addClassCleanup(patcher.stop)
+
+    def setUp(self):
+        self.module_root = tempfile.mkdtemp(prefix='odoo-test-module-', dir=self.addons_path)
+        self.module_name = os.path.basename(self.module_root)
+
+    def write_manifest(self, **overrides):
+        mandatories = {'name': f'Temp {self.module_name}', 'license': 'MIT'}
+        with open(opj(self.module_root, '__manifest__.py'), 'w') as manifest_fd:
+            manifest_fd.write(str(dict(mandatories, **overrides)))
```

Comment:
These three lines are only used once, and the method is always called without arguments.
Simplify and move those lines into the corresponding test method.

BTW the name `manifest_fd` is confusing because `open()` returns a _file_, not a _file descriptor_.
Why not `manifest_file`?  Are you nostalgic about C?  :wink:

Reply:
~/P/cpython$ rg 'with open' | grep -o 'as .*$' | sort | uniq -c | sort -n | tail -n 10
      7 as infile:
      7 as src:
      8 as output:
     10 as stream:
     12 as chan:
     20 as testfile:
     39 as fobj:
     69 as file:
    273 as fp:
    678 as f:

I was sure `fd` was the one commonly used. Changing it to `file`...